src/vcpkg/commands.ci.cpp

```diff
@@ -299,19 +299,22 @@ namespace vcpkg::Commands::CI
         const std::vector<FullPackageSpec>& specs,
         IBinaryProvider& binaryprovider,
         const Dependencies::CreateInstallPlanOptions& serialize_options,
+        Triplet target_triplet,
         Triplet host_triplet)
     {
         auto ret = std::make_unique<UnknownCIPortsResults>();
 
         auto is_excluded = [&](const PackageSpec& spec) -> bool {
+            bool excluded = false;
             if (spec.triplet() == host_triplet)
             {
-                return Util::Sets::contains(host_exclusions, spec.name());
+                excluded = excluded || Util::Sets::contains(host_exclusions, spec.name());
             }
-            else
+            if (spec.triplet() == target_triplet)
             {
-                return Util::Sets::contains(exclusions, spec.name());
+                excluded = excluded || Util::Sets::contains(exclusions, spec.name());
             }
+            return excluded;
```

Comment:
My optimizer brain hates mixing the control flow domain and the value domain. (Saying 'perf doesn't matter here leave me alone Bill' is a totally OK response :))

How about:

```
auto&& triplet = spec.triplet();
auto&& name = spec.name();
return (triplet == host_triplet && Util::Sets::contains(host_exclusions, name))
    || (triplet == target_triplet && Util::Sets::contains(exclusions, name));
```

Reply:
perf doesn't matter here leave me alone bill ;)

Reply:
To be honest I find the return expression a bit clearer too but 🤷‍♂️